projects/plugins/jetpack/modules/woocommerce-analytics/classes/class-jetpack-woocommerce-analytics-universal.php

```diff
@@ -194,11 +195,39 @@ public function capture_product_view() {
 		);
 	}
 
+	/**
+	 * Get the selected shipping option for a cart item. If the name cannot be found in the options table, the method's
+	 * ID will be used.
+	 *
+	 * @param string $cart_item_key the cart item key.
+	 *
+	 * @return mixed|void
+	 */
+	public function get_shipping_option_for_item( $cart_item_key ) {
+		$packages         = wc()->shipping()->get_packages();
+		$selected_options = wc()->session->get( 'chosen_shipping_methods' );
+		foreach ( $packages as $package_id => $package ) {
+			foreach ( $package['contents'] as $package_item ) {
+				if ( $package_item['key'] === $cart_item_key ) {
+					$selected_rate_id = $selected_options[ $package_id ];
```

Comment:
Same as above, should we be defensive here?

Reply:
Done in f7e34f92b7c6cf9858d7f35bd520262e4dfa89ec thanks!